ksqldb-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -958,6 +963,25 @@ private KsqlConfig(final ConfigGeneration generation, final Map<?, ?> props) {
     this.ksqlStreamConfigProps = buildStreamingConfig(streamsConfigDefaults, originals());
   }
 
+  private static Set<String> streamTopicConfigNames() {
+    final ImmutableSet.Builder<String> configs = ImmutableSet.builder();
+
+    Arrays.stream(TopicConfig.class.getDeclaredFields())
```

Comment:
heh.. this is a neat trick. I guess there's no other option to get these configs without reflection and just manually adding them?

Reply:
Right. I was thinking of manually creating the list, but TopicConfig can change in the future, and we'll have to adjust every time it changes. I'm not a fan of reflection, but I like the magic on it. This magic helps to forget about changes in TopicConfig as long as they stay public.